### PR TITLE
cleanup: git-extract uses sync_branch_steps

### PR DESCRIPTION
**Pull request by charlierudolph**
@kevgo @allewun

### CODE REVIEW

File: features/git-extract/cherry_pick_conflict/without_remote_origin.feature [hunk 1]
```diff
@@ -38,10 +37,9 @@ Feature: git extract: resolving conflicts between main branch and extracted comm
     Then it runs the commands
       | BRANCH   | COMMAND                 |
       | refactor | git cherry-pick --abort |
-      |          | git checkout main       |
-      | main     | git branch -d refactor  |
       |          | git checkout feature    |
-      | feature  | git stash pop           |
+      | feature  | git branch -D refactor  |
```
Comment (charlierudolph) [inline features/git-extract/cherry_pick_conflict/without_remote_origin.feature]: have to use force here as refactor isn't complete merged into HEAD, which is feature at the moment.

File: features/git-extract/cherry_pick_conflict/without_remote_origin.feature [hunk 2]
```diff
@@ -20,8 +20,7 @@ Feature: git extract: resolving conflicts between main branch and extracted comm
     Then it runs the commands
       | BRANCH   | COMMAND                                      |
       | feature  | git stash -u                                 |
-      |          | git checkout main                            |
-      | main     | git checkout -b refactor main                |
+      |          | git checkout -b refactor main                |
```
Comment (kevgo) [inline features/git-extract/cherry_pick_conflict/without_remote_origin.feature]: Nice cleanup!

File: src/helpers/git_helpers/branch_helpers.sh [hunk 3]
```diff
@@ -34,7 +34,9 @@ function create_and_checkout_branch {
 function delete_local_branch {
   local branch_name=$1
   local op="d"
-  if [ "$2" == "force" ]; then op="D"; fi
+  if [ "$2" == "force" ] || [ "$(delete_local_branch_needs_force "$branch_name")" = true ];
+    then op="D";
```
Comment (kevgo) [inline src/helpers/git_helpers/branch_helpers.sh]: Either move `then` to the end of the last line,
or don't indent it here.

```bash
if [ ...]
then
  op="D"
fi

**Comment by kevgo** [comment]
:ship: Cool structural dry up!

**Pull request closed without merging**